### PR TITLE
[Arctic-1347][Core]The position-delete written by iceberg directly through the base table cannot take effect when read by Arctic

### PR DESCRIPTION
**Pull request by shidayang**
## Why are the changes needed?
fix: #1347 

## Brief change log
Fix BasicArcticFileScanTask logic.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/scan/BasicArcticFileScanTask.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public BasicArcticFileScanTask(
     this.posDeleteFiles = posDeleteFiles == null ? Collections.emptyList() :
         posDeleteFiles.stream().filter(s -> {
           DataTreeNode node = FileNameGenerator.parseFileNodeFromFileName(s.path().toString());
-          return node.index() == baseFile.node().index() && node.mask() == baseFile.node().mask();
+          return baseFile.node().isSonOf(node) || baseFile.node().equals(node);
         }).collect(Collectors.toList());
```
Comment (baiyangtx) [inline core/src/main/java/com/netease/arctic/scan/BasicArcticFileScanTask.java:62]: I can't understand the logical, plz make a simple description

Comment (shidayang) [reply core/src/main/java/com/netease/arctic/scan/BasicArcticFileScanTask.java:62]: position-delete must be data file parent node or the same node

**Review by baiyangtx — APPROVED** [review]
LGTM
